src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const male = century
+    ? people.filter(person => person.sex === 'm'
+    && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const maleAge = male.map(person => person.died - person.born);
+
+  const averageAge = maleAge.reduce((a, b) => a + b) / maleAge.length;
+
+  return averageAge;
```

Comment:
Move this logic to the separated function and reuse it in your main functions to avoid duplication